src/solver/conv_hip_implicit_gemm_f16f8f16_bwd_xdlops.cpp

```diff
@@ -0,0 +1,368 @@
+        return true;
+    }
+    if((index + 1) < valid_kernels.size())
+    {
+        ++index;
+        kernel_id = valid_kernels[index];
+        return true;
+    }
+    else
+        return false;
+}
+
+bool PerformanceConfigHipImplicitGemmF16F8F16BwdXdlops::IsValidValue() const
+{
+    return index < valid_kernels.size();
+}
+
+bool PerformanceConfigHipImplicitGemmF16F8F16BwdXdlops::IsValid(
+    [[maybe_unused]] const ProblemDescription& problem) const
+{
+#if MIOPEN_BACKEND_HIP && MIOPEN_USE_COMPOSABLEKERNEL
+    const auto& y_cast_type = problem.GetOut().GetCastType();
+    const auto& w_cast_type = problem.GetWeights().GetCastType();
+    if(y_cast_type == miopenBFloat8 && w_cast_type == miopenFloat8)
+        return CheckIsSupportCKArgs<ck::half_t, ck::bf8_t, ck::f8_t>(problem);
+#endif
+    return false;
+}
+
+bool PerformanceConfigHipImplicitGemmF16F8F16BwdXdlops::operator==(
+    const PerformanceConfigHipImplicitGemmF16F8F16BwdXdlops& other) const
+{
+    return kernel_id == other.kernel_id;
+}
+
+PerformanceConfigHipImplicitGemmF16F8F16BwdXdlops
+ConvHipImplicitGemmF16F8F16BwdXdlops::GetDefaultPerformanceConfig(
+    const ExecutionContext&, const ProblemDescription& problem) const
+{
+    PerformanceConfigHipImplicitGemmF16F8F16BwdXdlops pp;
+    pp.HeuristicInit(problem);
+    return pp;
+}
+
+bool ConvHipImplicitGemmF16F8F16BwdXdlops::IsValidPerformanceConfig(
+    const ExecutionContext&,
+    const ProblemDescription& problem,
+    const PerformanceConfigHipImplicitGemmF16F8F16BwdXdlops& config) const
+{
+    return config.IsValid(problem);
+}
+
+PerformanceConfigHipImplicitGemmF16F8F16BwdXdlops
+ConvHipImplicitGemmF16F8F16BwdXdlops::Search(const ExecutionContext& ctx,
+                                             const ProblemDescription& problem,
+                                             const AnyInvokeParams& invoke_ctx) const
+{
+    return GenericSearch(*this, ctx, problem, invoke_ctx);
+}
+
+bool ConvHipImplicitGemmF16F8F16BwdXdlops::IsApplicable(
+    [[maybe_unused]] const ExecutionContext& ctx,
+    [[maybe_unused]] const ProblemDescription& problem) const
+{
+#if MIOPEN_BACKEND_HIP && MIOPEN_USE_COMPOSABLEKERNEL
+    if(miopen::IsDisabled(MIOPEN_DEBUG_F16F8F16_CONV_IMPLICIT_GEMM_HIP_BWD_XDLOPS{}))
+        return false;
+    if(miopen::IsEnabled(MIOPEN_DEBUG_CONVOLUTION_DETERMINISTIC{}))
+        return false;
+    if(problem.HasMixedDataTypes())
+        return false;
+    if(!problem.IsTensorsCasted())
+        return false;
+    if(!problem.IsDirectionBackwardData())
+        return false;
+    if(!problem.Is3d())
+        return false;
+    if(!problem.IsLayoutNHWC())
+        return false;
+    if(!problem.IsFp16())
+        return false;
+    if(!ck_utility::is_ck_whitelist(ctx.GetStream().GetDeviceName()))
+        return false;
+    const auto& yDesc = problem.GetOut();
+    const auto& wDesc = problem.GetWeights();
+    if(yDesc.GetCastType() && wDesc.GetCastType())
```

Comment:
> [Q] Why don't we check casting for dxDesc? PR description says "fp32->fp16". Or frameworks do not require us to compute to fp32 and then cast output to fp16, i.e. they only ask us to cast inputs from 16 to 8 bits? And this fp32->fp16 casting is simply inherent to the implementation?

We do not need to check from MIOpen side. The [backward data API](https://github.com/ROCmSoftwarePlatform/composable_kernel/blob/develop/include/ck/tensor_operation/gpu/device/device_grouped_conv_bwd_data_multiple_d.hpp) that CK provide to us only requests the compute type for output and weights, the input type casting is handled by the CK kernel.

Reply:
[Recommendation] @iq136boy Which means that fp32->fp16 casting of _output_ (dx tensor) is a detail of implementation, right? (Ditto other solvers.) This is fine so far. However it may worth explaining this in the description of the PR.

What I see:
- The kernels convert FP16 inputs to FP8 or BF8 internally. This improves performance at the cost of accuracy. The conversion is a parameter of convolution and expressed in terms of "tensor casting" of inputs.
- The kernels perform intermediate computations in FP32 domain and convert output to FP16. This is fairly common in MIOpen and does not require additional convolutions parameters.

I recommend updating the description of this PR with the explanation shown above.